### PR TITLE
feat: added flake8, black, isort, and pre-commit

### PR DESCRIPTION
**Pull request by Fagtoy**
Added flake8, black, isort, and pre-commit configurations. All flake8 errors are resolved. Local installation and running remain the same as for the backend repo.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[flake8]
+exclude=.git,__pycache__,docs/source/conf.py,old,build,dist,venv,__init__.py,./lib/*,./bin/*
```
Comment (richardzilincikPantheon) [inline .flake8:2]: We have pretty much all of our code in the bin directory, won't this stop it from being checked? (I'm not sure why we have everything inside bin, should probably be taken out or renamed to something that makes more sense.)

File: requirements.txt [hunk 2]
```diff
@@ -9,3 +9,15 @@ filelock==3.4.0
 redis==4.1.2
 pytest==7.0.1
 scipy==1.8.1
+pre-commit==2.20.0
```
Comment (richardzilincikPantheon) [inline requirements.txt:12]: I didn't notice this in the backend PR, but why do we want to add these as dependencies? If it's just for the github actions tests, it's probably better to put them in a separate file.

**Review by richardzilincikPantheon — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (Fagtoy) [reply requirements.txt:12]: moved all linting-related dependencies to a separate ```lint_requirements.txt``` file

Comment (SlavomirMazurPantheon) [reply requirements.txt:12]: If you moved them to the `lint_requirements.txt` they should be removed from this file

**Review by SlavomirMazurPantheon — APPROVED** [review]

**Pull request merged**